packages/core/scripts/admin-proposals/collateral.js

```diff
@@ -0,0 +1,325 @@
+  } else if (collateral) {
+    collaterals = collateral.split(",");
+  } else {
+    throw new Error("Must specify either --polygon or --collateral or both");
+  }
+
+  if (
+    (collaterals && collaterals.length !== fees.length) ||
+    (polygonCollaterals && polygonCollaterals.length !== fees.length)
+  ) {
+    throw new Error("all comma-delimited input strings should result in equal length arrays");
+  }
+
+  // Initialize Eth contracts by grabbing deployed addresses from networks/1.json file.
+  if (netId === HARDHAT_NET_ID) netId = PROD_NET_ID;
+  const whitelist = new web3.eth.Contract(AddressWhitelist.abi, _getContractAddressByName("AddressWhitelist", netId));
+  const store = new web3.eth.Contract(Store.abi, _getContractAddressByName("Store", netId));
+  const gasEstimator = new GasEstimator(
+    winston.createLogger({ silent: true }),
+    60, // Time between updates.
+    netId
+  );
+  await gasEstimator.update();
+  console.log(
+    `⛽️ Current fast gas price for Ethereum: ${web3.utils.fromWei(
+      gasEstimator.getCurrentFastPrice().toString(),
+      "gwei"
+    )} gwei`
+  );
+  const governor = new web3.eth.Contract(Governor.abi, _getContractAddressByName("Governor", netId));
+  const finder = new web3.eth.Contract(Finder.abi, _getContractAddressByName("Finder", netId));
+  const oracleAddress = await finder.methods
+    .getImplementationAddress(web3.utils.utf8ToHex(interfaceName.Oracle))
+    .call();
+  const oracle = new web3.eth.Contract(Voting.abi, oracleAddress);
+  const governorRootTunnel = new web3.eth.Contract(
+    GovernorRootTunnel.abi,
+    _getContractAddressByName("GovernorRootTunnel", netId)
+  );
+  if (polygonCollaterals) {
+    console.group("\nℹ️  Relayer infrastructure for Polygon transactions:");
+    console.log(`- Store @ ${polygon_store.options.address}`);
+    console.log(`- AddressWhitelist @ ${polygon_whitelist.options.address}`);
+    console.log(`- GovernorRootTunnel @ ${governorRootTunnel.options.address}`);
+    console.groupEnd();
+  }
+  console.group("\nℹ️  DVM infrastructure for Ethereum transactions:");
+  console.log(`- Store @ ${store.options.address}`);
+  console.log(`- AddressWhitelist @ ${whitelist.options.address}`);
+  console.log(`- Finder @ ${finder.options.address}`);
+  console.log(`- Oracle @ ${oracle.options.address}`);
+  console.log(`- Governor @ ${governor.options.address}`);
+  console.groupEnd();
+
+  if (!verify) {
+    console.group("\n🌠 Proposing new Admin Proposal");
+
+    const adminProposalTransactions = [];
+    console.group("\n Key to understand the following logs:");
+    console.log(
+      "- 🟣 = Transactions to be submitted to the Polygon contracts are relayed via the GovernorRootTunnel on Etheruem. Look at this test for an example:"
+    );
+    console.log("    - https://github.com/UMAprotocol/protocol/blob/master/packages/core/test/polygon/e2e.js#L221");
```

Comment:
Should we point this at a hardcoded commit hash or something in case this file changes?

Reply:
Great point, done.